_posts/no-orchstration.md

```diff
@@ -0,0 +1,57 @@
+
+In a nutshell, that's the way, a couple of years ago, in the context of an architectural review, I was presented with requirements for a booking system.
+
+## It oozes orchestration from every pore
+
+The team choice was to implement the system using an orchestration architectural style. For example, a `BookingService` uses commands to pilot the steps of the process across many different services, such as checkout, payments, customer management, booking, and many more. The following picture summarizes the described process:
+
+![booking service orchestrator diagram](/img/posts/no-orchstration/orchestrator.png){:class="img-fluid mx-auto d-block"}
+
+Let me be clear. The system was alive and kicking. It had been selling tickets for the last few years (and still is) and didn't have any significant issues. The review was in the context of some concerns the team had with recent changes they were trying to implement.
+
+Without going into the details of the changes, the team was concerned by things like:
+
+- to add a new payment type, they needed to change the booking service
+- to change the way customers are created/handled, they needed to change the booking service
+- to add groups tickets, they needed to change the booking service
+
+And many more. Every single change they were making to the business process had to go through the booking service. The booking service was a controlling actor in the system. It was acting like a centralizing entity forcing all changes to go through it.
+
+In other words, the booking service violates the single responsibility principle (SRP). It has too many responsibilities. Of course, one could argue that we could design a booking service composed of many classes, each with a single goal. But, even if that's true, it doesn't change the reality that the booking service centralizes all the duties in a single place.
+
+## Why do we think we need a booking service?
+
+We think we need a booking service that orchestrates the process because we fall into the trap of modeling the user mental model. [We model the user information and data representation instead of the behaviors that manipulate those data](https://milestone.topics.it/2021/02/02/do-not-trust-the-user-mental-model.html). Users describe the booking process in such a way that emphasizes the centrality of the booking service. Not to mention that they continuously talk about how information and data are displayed. For example, by dictating that there is the need for a dashboard showing the state of the ongoing business process. ViewModel Composition techniques can solve the [dashboard dilemma](link to composing emails).
```

Comment:
```suggestion
We think we need a booking service that orchestrates the process because we fall into the trap of modeling the user mental model. [We model the user information and data representation instead of the behaviors that manipulate those data](https://milestone.topics.it/2021/02/02/do-not-trust-the-user-mental-model.html). Users describe the booking process in such a way that emphasizes the centrality of the booking service. Not to mention that they continuously talk about how information and data are displayed. For example, by dictating that there is the need for a dashboard showing the state of the ongoing business process. ViewModel Composition techniques can solve the [dashboard dilemma](https://milestone.topics.it/2021/07/02/you-want-to-compose-emails-right.html).
```